File: webapp/assets/js/publicacoes.js
$('#nova-publicacao').on('submit', criarPublicacao)

function criarPublicacao(e) {
    e.preventDefault()

    console.log("titulo:", $('#titulo').val())
    console.log("conteudo:", $('#conteudo').val())


    $.ajax({
        url: "/publicacoes",
        method: "POST",
        contentType: "application/json",
        data: JSON.stringify({
            titulo: $('#titulo').val(),
            conteudo: $('#conteudo').val()
        })
    }).done(function () {
        window.location = "/home"
    }).fail(function () {
        alert("erro ao criar")
    })
}




$(".curtir-publicacao").on('click', curtirPublicacao)
function curtirPublicacao(e) {

    e.preventDefault()

    const elementoClicado = $(e.target)
    const publicacaoId = elementoClicado.closest('[data-publicacao-id]').data('publicacao-id')

    elementoClicado.prop('disabled', true)

    $.ajax({
        url: `/publicacoes/${publicacaoId}/curtir`,
        method: "POST",
    }).done(function () {

        const contador = elementoClicado.find('.contador-curtidas')
        const quantidade = parseInt(contador.text())

        contador.text(quantidade + 1)

    }).fail(function () {
        alert('erro ao curtir')
    }).always(function () {
        elementoClicado.prop('disabled', false)

    })

}

File: webapp/src/models/Publicacoes.go
package models

import "time"

type Publicacao struct {
	ID        uint64    `json:"id,omitempty"`
	Titulo    string    `json:"titulo,omitempty"`
	Conteudo  string    `json:"conteudo,omitempty"`
	AutorID   uint64    `json:"autor_id,omitempty"`
	AutorNick string    `json:"autor_nick,omitempty"`
	Curtidas  uint64    `json:"curtidas"`
	CriadaEm  time.Time `json:"criadaEm,omitempty"`
}
